src/sentry/migrations/0024_auto_20191230_2052.py

```diff
@@ -55,6 +55,18 @@ def _attach_related(_events):
             event.group = groups.get(event.group_id)
         eventstore.bind_nodes(_events, "data")
 
+    def _attach_timestamp(_events):
+        """
+        When migrating old data from Sentry 9.0.0 to 9.1.2 to 10 in rapid succession, the event timestamp may be missing.
+        This function exists to attach the timestamp if its missing
+        """
+        for event in _events:
+            if not event.data.data.get("timestamp"):
+                # Python 2.7 way to convert datetime to timestamp.
```

Comment:
FYI we are working towards a migration to Python 3.6 and beyond so I hope this works in them too?

Reply:
Yes, this works going forward as well. The main consideration is that python 3.3+ has a function that converts a datetime to a timestamp without doing the ugliness below.

Reply:
I think you can leverage the `to_timestamp` helper here (and remove the python 2.7 comment) https://github.com/getsentry/sentry/blob/master/src/sentry/utils/dates.py#L15-L23

Reply:
Ah perfect! I'll update now and then run through a test cycle